server/tilesource/cache.py

```diff
@@ -87,7 +87,8 @@ def __call__(cls, *args, **kwargs):
         instance = cache.get(key, _MARKER)
         if instance is _MARKER:
             instance = super(LruCacheMetaclass, cls).__call__(*args, **kwargs)
-            cache.put(key, instance)
+            if key is not None:
```

Comment:
Why do we need this? In theory, ``keyFunc`` could return ``None`` as its key value (which is why we instead use ``_MARKER`` to represent the absence of a key).